crates/rattler_installs_packages/src/index/package_database.rs

```diff
@@ -232,7 +233,35 @@ impl PackageDb {
     ) -> miette::Result<Wheel> {
         // TODO: add support for this currently there are not saved
         if artifact_info.is::<STree>() {
-            miette::bail!("STree artifacts are not supported");
+            if let Some(builder) = builder {
+                let stree_name = artifact_info
+                    .filename
+                    .as_inner::<STreeFilename>()
+                    .unwrap_or_else(|| {
+                        panic!(
+                            "the specified artifact '{}' does not refer to type requested to read",
+                            artifact_info.filename
+                        )
+                    });
+
+                let response = super::direct_url::fetch_artifact_and_metadata_by_direct_url(
+                    &self.http,
+                    stree_name.distribution.clone(),
+                    artifact_info.url.clone(),
+                    builder,
+                )
+                .await?;
+
+                let stree = response
+                    .artifact
+                    .as_stree()
+                    .expect("the request artifact does not refer to stree");
+
+                let built = builder.build_wheel(stree).await.into_diagnostic();
+                return built;
+            } else {
+                miette::bail!("cannot build wheel without a wheel builder");
+            }
         }
 
         // Try to build the wheel for this SDist if possible
```

Comment:
I might be missing something but one of the original problems, was that if a `wheel` or `sdists` was referenced by file, this call below will fail.

Reply:
Actually, just tested this and it still does:

```bash
$ cargo r "miniblack @ file:///Users/tdejager/development/prefix/rattler_installs_packages/test-data/wheels/miniblack-23.1.0-py3-none-any.whl" --install-into /tmp/bla2

# errors with:
installing: miniblack - 23.1.0
2024-02-08T08:45:22.764833Z  INFO rattler_installs_packages::index::http: executing request url=file:///Users/tdejager/development/prefix/rattler_installs_packages/test-data/wheels/miniblack-23.1.0-py3-none-any.whl cache_mode=Default
  × Request error: builder error for url (file:///Users/tdejager/development/prefix/rattler_installs_packages/test-data/wheels/miniblack-23.1.0-py3-none-any.whl): URL scheme is not allowed
  ├─▶ builder error for url (file:///Users/tdejager/development/prefix/rattler_installs_packages/test-data/wheels/miniblack-23.1.0-py3-none-any.whl): URL scheme is not allowed
  ╰─▶ URL scheme is not allowed
```

Reply:
yes that's true - I've missed this and spotted during direct_url.json development.
Will fix it and add a test